src/Microsoft.Dnx.Runtime/Compilation/ICompilationSession.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Microsoft.Dnx.Runtime.Compilation
+{
+    public interface ICompilationSession : IDisposable
+    {
+        event Action<string> OnInputFileChanged;
+
+        Assembly CompileAndLoadProject(Project project, CompilationTarget target, IAssemblyLoadContext loadContext);
```

Comment:
Not just the name the fact that the ICompilationSession has a Load on it. I don't get the design.